### PR TITLE
Feat/model save

### PR DESCRIPTION
**Pull request by SunCreation**
During training, model is saved whenever it attain the highest score.
And we can load model by using 
```python
model.load("model_path.pt")
```

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -172,6 +172,14 @@ def train(index,args):
                             ).per_device_loader(device)
                 score = evaluate(model=model, dataloader=val_dataloader, tokenizer=tokenizer, gpt_tokenizer=gpt_tokenizer, wandb=wandb)
                 wandb.log({"val_"+i:j for i,j in score.items()})
+                
+                with open("score") as f:
```
Comment (YejinHwang909) [inline train.py:176]: @SunCreation
minor, how about adding an extension `txt`
```suggestion
                with open("score.txt") as f:
```

File: train.py [hunk 2]
```diff
@@ -216,6 +227,9 @@ def train(index,args):
     config.text_encoder = Config.from_json("text_encoder_config.json")
 
     model_ = MetaLM(config)
+    if config.load_path: model_.load(config.load_path)
+    with open("score","w") as f:
```
Comment (YejinHwang909) [inline train.py:231]: same comment!

**Review by YejinHwang909 — APPROVED** [review]
GOOD! 😆

File: model.py [hunk 3]
```diff
@@ -25,7 +25,6 @@ def __init__(self, config):
         # GPI is Semi-Causal LM
         self.GPI = GPT2ForTokenClassification.from_pretrained('skt/kogpt2-base-v2',output_hidden_states=True,num_labels=config.num_labels)
         self.p_tunning = config.p_tunning
-        self.classifier = nn.Linear(768,7)
```
Comment (YejinHwang909) [inline model.py:28]: good!

File: model.py [hunk 4]
```diff
@@ -122,3 +121,29 @@ def forward(self, inputs, order=False):
...
         return outputs
 
+    def save(self, PATH: str, optimizer: torch.optim.Optimizer = None) -> None:
+        model_for_save = {
+            "SpecialLayer": self.SpecialLayer.state_dict(),
+            "prefix_encoder": self.prefix_encoder.state_dict(),
+            "classifier": self.GPI.classifier.state_dict(),
+        }
+        model_for_save.update(**self.config.__dict__)
+        model_for_save['text_encoder'] = model_for_save['text_encoder'].__dict__
```
Comment (YejinHwang909) [inline model.py:131]: @SunCreation 
text_encoder and wav_encoder are pre-trained model, so is it necessary to save them?

Comment (YejinHwang909) [reply model.py:131]: I think we can just give a model name(Klue/RoBERTa-large, whisper-base)? 
Please consider it. It is up to you!

Comment (SunCreation) [reply model.py:131]: @YejinHwang909 
haha.. Actually, they are just Config class.
I also thought it's miss naming.😭

File: train.py (hunk 1 above)
Comment (SunCreation) [reply train.py]: Thank you for checking :)

**Comment by SunCreation** [comment]
Thank you for delicately checking!!

**Pull request merged**